src/passes/OptimizeInstructions.cpp

```diff
@@ -2586,10 +2608,8 @@ struct OptimizeInstructions
       Binary *bx, *by;
       Expression *x, *y;
       Const *cx, *cy;
-      if (matches(curr,
-                  binary(OrInt32,
-                         binary(&bx, any(&x), ival(&cx)),
-                         binary(&by, any(&y), ival(&cy)))) &&
+      if (matches(curr->left, binary(&bx, any(&x), ival(&cx))) &&
+          matches(curr->right, binary(&by, any(&y), ival(&cy))) &&
```

Comment:
Where did the check for OrInt32 go?

Reply:
It was earlier: https://github.com/WebAssembly/binaryen/blob/9b6f92165072d46331533b767bf9ec26c9ae1b2d/src/passes/OptimizeInstructions.cpp#L2549